pubtools/_quay/operator_pusher.py

```diff
@@ -453,25 +453,97 @@ def build_index_images(self):
 
         """
         iib_results = {}
+        repos_opted_in = {}
+        items_opted_in = {}
+        failed_items = {}
+
+        # We need to load pyxis resolved versions at this point
+
+        versions_mapping = self.version_items_mapping  # noqa: F841
+
+        for item in self.push_items:
+            for repo in item.metadata["tags"]:
+                if repo not in repos_opted_in:
+                    repos_opted_in[repo] = pyxis_get_repo_metadata(repo, self.target_settings).get(
+                        "fbc_opt_in", False
+                    )
+            item_fbc_opt_in = [repos_opted_in[repo] for repo in item.metadata["tags"]]
+            if not (all(item_fbc_opt_in) or not any(item_fbc_opt_in)):
+                failed_items[id(item)] = True
+                item.add_error("NOTPUSHED", "Inconsistent fbc opt in")
+                LOG.error("Inconsistent fbc opt in for item {i}".format(i=item))
+                continue
+            elif all(item_fbc_opt_in):
+                items_opted_in[id(item)] = True
+            else:
+                items_opted_in[id(item)] = False
+
+            ocp_versions = item.metadata["com.redhat.openshift.versions"]
+            if (
+                [
+                    version
+                    for version in self.ocp_versions_resolved[ocp_versions]
+                    if tuple([int(x) for x in version.replace("v", "").split(".")]) < (4, 10)
+                ]
+                and [
+                    version
+                    for version in self.ocp_versions_resolved[ocp_versions]
+                    if tuple([int(x) for x in version.replace("v", "").split(".")]) > (4, 9)
+                ]
+                and items_opted_in[id(item)]
+            ):
+                item.add_error(
+                    "INVALIDFILE",
+                    "Cannot push item to index image "
+                    "as it supports both <= 4.9 and >= 4.10 and is opted in FBC: {item}".format(
+                        item=item
+                    ),
+                )
+                LOG.error(
+                    "Cannot push item to index image "
+                    "as it supports both <= 4.9 and >= 4.10 and is opted in FBC: {item}".format(
```

Comment:
sqlite images are until 4.10 and fbc images start from 4.11, so this should be 
"as it supports both <= 4.10 and >= 4.11 and is opted in FBC"

Reply:
According to Yash, it was 4.9, 4.10, 4.11. Are you sure it should fail when bundle is for v4.10, v4.11 and is opted in?

Reply:
fixed